controllers/cloudstackfailuredomain_controller.go

```diff
@@ -107,6 +114,43 @@ func (r *CloudStackFailureDomainReconciliationRunner) Reconcile() (retRes ctrl.R
 	return ctrl.Result{}, nil
 }
 
+func (r *CloudStackFailureDomainReconciliationRunner) SetOwnerReferenceToEndpointSecret() error {
+	secret, err := r.GetSecretForFailureDomain(&r.ReconciliationSubject.Spec)
+	if err != nil {
+		return err
+	}
+	if MaySetOwnerReference(secret, r.CAPICluster) {
+		return r.Patcher.Patch(r.RequestCtx, secret)
+	}
+	return nil
+}
+
+func MaySetOwnerReference(secret *corev1.Secret, cluster *clusterv1.Cluster) bool {
+	needSetOwner := false
+	for key, value := range secret.GetLabels() {
+		if strings.EqualFold(key, clusterv1.ClusterLabelName) && strings.EqualFold(value, cluster.Name) {
+			needSetOwner = true
+			for _, ref := range secret.GetOwnerReferences() {
+				if strings.EqualFold(ref.Kind, "Cluster") {
+					needSetOwner = false
+					break
+				}
+			}
+			break
+		}
+	}
+	if needSetOwner {
+		secret.OwnerReferences = append(secret.OwnerReferences,
+			metav1.OwnerReference{
+				Name:       cluster.Name,
+				Kind:       cluster.Kind,
+				APIVersion: cluster.APIVersion,
+				UID:        cluster.UID,
```

Comment:
I'm not a fan of modifying input parameters. Can we just return `needSetOwner` and modify the secret in the client?